guides/release/tutorial/18.md

```diff
@@ -0,0 +1,192 @@
+import Component from '@glimmer/component';
+import ENV from 'super-rentals/config/environment';
+
+export default class MapComponent extends Component {
+  get token() {
+    return encodeURIComponent(ENV.MAPBOX_ACCESS_TOKEN);
+  }
+}
+```
+
+Here, we are importing the access token from the config file and returning it from a `token` *[getter](TODO: link to getter)*. This allows us to access our token as `this.token` on the instance of our component, including on the component's template. It is also important to [URL-encode](TODO: link to URL-encode) the token, just in case it contains any special characters that are not URL-safe.
+
+Now, let's move from the JavaScript file to the template:
+
+```handlebars {data-filename="app/components/map.hbs}
+<div class="map">
+  <img
+    alt="Map image at coordinates {{@lat}},{{@lng}}"
+    ...attributes
+    src="https://api.mapbox.com/styles/v1/mapbox/streets-v11/static/{{@lng}},{{@lat}},{{@zoom}}/{{@width}}x{{@height}}@2x?access_token={{this.token}}"
+    width={{@width}} height={{@height}}
+  >
+</div>
+```
+
+First, we have a container element for styling purposes.
+
+Then we have an `<img>` tag to request and render the static map image from Mapbox.
+
+We generated a reasonable default value for the `alt` attribute` based on the value of `{{@lat}}` and `{{@lng}}`. These are the *[arguments](TODO: link to arguments)* that the invoker of the `<Map>` component would need to specify when invoking it.
+
+By *[parameterizing](TODO: link to parameterizing)* our component using arguments, we made a reusable component that can be invoked from different parts of the app and customized to meet the needs for those specific contexts.
+
+We have already seen this in action when using the `<LinkTo>` component earlier; we had to specify a `@route=...` argument so that it knew what page to navigate to.
+
+You may also notice that we are directly *[interpolating](TODO: link to interpolating)* values into the `alt` attribute's value. Ember will automatically concatenate these interpolated values into a final string value for us, including doing any necessary escaping, if appropriate.
+
+Next, we used `...attributes` to allow the invoker to further
+customize the `<img>` tag, such as passing extra attributes such
+as `class`, as well as *[overriding](TODO: link to overriding)* our default `alt` attribute with a more specific or human-friendly one.
+
+*The ordering is important here!* Ember applies the attributes in the order that they appear. By assigning the `alt` attribute first, *before* `...attributes` is applied, we are explicitly allowing the invoker to optionally pass a different value for the `alt` attribute.
+
+Since the passed-in `alt` attribute (if any exists) will appear _after_ ours, it will override the value we specified. On the other hand, it is important that we assign `src`, `width`, and `height` after `...attributes`, so that they don't get accidentally overwritten by the invoker.
```

Comment:
```suggestion
Since the passed-in `alt` attribute (if any) will appear _after_ ours, it will override the value we specified. On the other hand, it is important that we assign `src`, `width`, and `height` after `...attributes`, so we don't actually overwrite them &mdash; those attributes should always be specified by the component.
```